dandi/dandiapi.py

```diff
@@ -325,24 +349,82 @@ class Version(APIBase):
     modified: datetime
 
 
-class RemoteDandiset(APIBase):
+class RemoteDandiset:
     """
     Representation of a Dandiset (as of a certain version) retrieved from the
     API
     """
 
-    client: "DandiAPIClient"
-    identifier: str
-    created: datetime
-    modified: datetime
-    #: The version in question of the Dandiset
-    version: Version
-    most_recent_published_version: Optional[Version]
-    draft_version: Version
+    def __init__(
+        self,
+        client: DandiAPIClient,
+        identifier: str,
+        version: Union[str, Version, None],
```

Comment:
there should be a docstring which would describe what happens if `version=None`

Reply:
oh, from right below https://github.com/dandi/dandi-cli/pull/676/files#r673408547 I see that intention is to have these classes not really to be instantiated by "users" (who might be developers) directly... hm... need to digest